[SYCL][DOC] Add Extension for FPGA task_sequence (#6348)
intel/llvm: sycl/doc/extensions/proposed/sycl_ext_intel_fpga_kernel_interface_properties.asciidoc

```c++
namespace sycl::ext::intel::experimental {

> enum class streaming_interface_options_enum : /* unspecified */ {

Reviewer: Why mention that this class inherits from an unspecified base class?  Does this affect the user in any way?  It seems like an implementation detail to me but maybe I'm missing something.

Reviewer 2: I checked the previous commits, and it is unclear that this was added as a response to a specific comment, so I am removing it.

 If the property parameter (N) is not specified, the default value is `-1`.
 Valid values for `N` are values greater than or equal to `-1`.

|`fpga_cluster`

> |The `fpga_cluster` property adds a request, to the extent possible, for the

Reviewer: ```suggestion
|The `fpga_cluster` property is a request, to the extent possible, for the
```
or perhaps "indicates a request" but "adds a request" sounds awkward.

 compiler to implement statically-scheduled clusters using the specified

>  clustering method. The following values are supported:

Reviewer: ```suggestion
 clustering method for all clusters in the annotated kernel.  The following values are supported:
```

 * `stall_enable`: Directs the compiler to prefer generating an enable signal to
    freeze the cluster when the cluster is stalled whenever possible.

 * `stall_free`: Directs the compiler to prefer using an exit FIFO to hold
    output data from clusters when the cluster is stalled whenever possible.

>  *Note*: Not all clusters can be implemented with the provided preference.

Reviewer: This is a weird way to say it as it mixes general (all theoretical clusters in existence) with specific (the preference provided by a single instance of the property).  I think a cleaner way to state this would be:

```suggestion
 *Note*: Some clusters cannot be implemented with some cluster types so the request isn't guaranteed to be followed.  
```

 If the `fpga_cluster` property is not specified, the default behavior is
 equivalent to one of the values listed above, but the choice is implementation

>  defined.

Reviewer: We definitely shouldn't say this.  If the property is not specified we want to reserve the right to decide on a cluster-by-cluster basis whereas this text implies that we have to pick one type wherever possible and document that choice in our docs.  I'd remove this sentence entirely.
